CRM/Utils/System.php

```diff
@@ -1416,8 +1416,14 @@ public static function civiExit($status = 0) {
   public static function flushCache() {
     // flush out all cache entries so we can reload new data
     // a bit aggressive, but livable for now
-    $cache = CRM_Utils_Cache::singleton();
-    $cache->flush();
+    CRM_Utils_Cache::singleton()->flush();
+    if (Civi\Core\Container::isContainerBooted()) {
+      Civi::cache('settings')->flush();
```

Comment:
So @totten what strikes me here is what if my extension uses a different key - it will not now be flushed & I have 'no say' in that...